src/AWS.Deploy.Orchestration/CdkProjectHandler.cs

```diff
@@ -168,27 +196,55 @@ public async Task<bool> DetermineIfCDKBootstrapShouldRun()
             return false;
         }
 
-        private async Task CheckCdkDeploymentFailure(CloudApplication cloudApplication, DateTime deploymentStartDate)
+        private async Task<string> RetreiveStackId(CloudApplication cloudApplication, CancellationToken cancellationToken)
+        {
+            // Do an initial delay to avoid a race condition of the status being checked before the deployment has kicked off.
+            await Task.Delay(TimeSpan.FromSeconds(3));
+
+            await WaitUntilHelper.WaitUntil(async () =>
+            {
+                try
+                {
+                    var stack = await _awsResourceQueryer.GetCloudFormationStack(cloudApplication.Name);
```

Comment:
Declare the `stack` variable outside if the `WaitUtil`. Then you don't have to an extra call to `GetCloudFormationStack` after the wait call.

Reply:
Done